core/mock_test.go

```diff
@@ -6,14 +6,31 @@ import (
 	rpctest "github.com/celestiaorg/celestia-core/rpc/test"
 )
 
-// StartMockNode starts a mock core node background process and returns it.
+// StartMockNode starts a mock Core node background process and returns it.
 func StartMockNode() *node.Node {
 	app := kvstore.NewApplication()
 	app.RetainBlocks = 10
-	return rpctest.StartTendermint(app, rpctest.SuppressStdout)
+	return rpctest.StartTendermint(app, rpctest.SuppressStdout, rpctest.RecreateConfig)
 }
 
-// MockClient returns a started Mock Core Client.
-func MockClient() Client {
+// MockEmbeddedClient returns a started mock Core Client.
+func MockEmbeddedClient() Client {
 	return NewEmbeddedFromNode(StartMockNode())
 }
+
+// StartRemoteClient returns a started remote Core node process, as well its
+// mock Core Client.
+func StartRemoteClient() (*node.Node, Client, error) {
```

Comment:
another optional nit: it is common in Go for test helpers not to return an error and instead to pass T with call of Fatal in case something breaks, so the function would look like:
```StartRemoteClient(t *testing.T) *node.Node, Client```

Reply:
yeah I know but I think we would use this function in other non-unit tests later on that may not use the `testing` pkg.